src/Application/Cli/TravisAutoDeployCommand.php

```diff
@@ -101,19 +102,21 @@ protected function execute(InputInterface $input, OutputInterface $output)
             return;
         }
 
-        // set git user data
-        $output->writeln('<info>Setting up git user</info>');
-        $this->commandRunner->run('git config --global user.name "${GIT_NAME}"');
-        $this->commandRunner->run('git config --global user.email "${GIT_EMAIL}"');
-
         // getting current php version to only deploy once
+        // @deprecated this "hack" will be deleted in next major release, see Travis config file for new way to handle this
         $currentPhpVersion = getenv('TRAVIS_PHP_VERSION');
         if ($input->getOption('php-version') != $currentPhpVersion) {
+            trigger_error('Deprecated option "php-version" called.', E_USER_NOTICE);
             $output->writeln('<comment>This version of the documentation is already deployed</comment>');
 
             return;
         }
 
+        // set git user data
+        $output->writeln('<info>Setting up git user</info>');
+        $this->commandRunner->run('git config --global user.name "${GIT_NAME}"');
+        $this->commandRunner->run('git config --global user.email "${GIT_EMAIL}"');
```

Comment:
Because I thought that we should not change the git user unless we can really deploy the docs as per the `if` call just above...